### PR TITLE
fix iptables mode hard code in e2e test

### LINKED ISSUE #46078: [e2e test] Why `should check kube-proxy urls` case always assume proxy mode is iptables?
**What?**

When I run e2e test in my local machine with kube-proxy running in **userspace** mode. I find test case:

"should check kube-proxy urls" in `test/e2e/networking.go` will try to hit `/proxyMode` and assume it is **iptables** mode by default. see https://github.com/kubernetes/kubernetes/blob/master/test/e2e/networking.go#L86

If it's not easy to pass in kube-proxy mode in e2e test framework, I wonder why check proxy mode and hard code it to be **iptables**?

### PR DESCRIPTION
**Pull request by m1093782566**
Fixes #46078

### CODE REVIEW

File: test/e2e/networking.go [hunk 1]
```diff
@@ -83,7 +83,6 @@ var _ = framework.KubeDescribe("Networking", func() {
 
 		By("checking kube-proxy URLs")
 		config.GetSelfURL(ports.ProxyHealthzPort, "/healthz", "200 OK")
-		config.GetSelfURL(ports.ProxyStatusPort, "/proxyMode", "iptables") // the default
```
Comment (wojtek-t) [inline test/e2e/networking.go]: We definitely shouldn't just remove it. We should come up with a better way to check it, but removing it is not a solution.

Comment (m1093782566) [inline test/e2e/networking.go]: I mean we can already check kube-proxy process's status by visiting `/healthz` endpoint. Why do we need to visit `proxyMode` endpoint again?

Comment (wojtek-t) [inline test/e2e/networking.go]: We want to know whether this endpoint is actually exposed or not.

Comment (m1093782566) [inline test/e2e/networking.go]: Hmmm. What about check the status code instead of response content?

**Review by cmluciano — COMMENTED** [review]
I'm guessing this would break pre-existing tests

@dcbw Is the usermode proxy in deprecation mode or do we just test that elsewhere?

**Comment by cmluciano** [comment]
Checking tests

@k8s-bot ok to test

**Comment by m1093782566** [comment]
@cmluciano 

We will support more proxy modes(apart from userspace and iptables) in the future, right? Actually, there is a work of support ipvs mode for kube-proxy, see #46580

Of course, I will keep that case and try to come up with a better way.

**Comment by m1093782566** [comment]
@wojtek-t @cmluciano I want to try 

```
curl -o /dev/null -s -w %{http_code} http://127.0.0.1:10249/proxyMode
```

to check the status code instead of response body.

Would do you think about the idea? Thanks?

**Comment by wojtek-t** [comment]
@freehan @bowei WDYT ^^

**Comment by bowei** [comment]
@wojtek-t checking 200 ok seems reasonable given people are working on more than iptables mode.

**Comment by m1093782566** [comment]
@wojtek-t @bowei @cmluciano 

PR updated, PTAL.

File: test/e2e/framework/networking_utils.go [hunk 2]
```diff
@@ -273,12 +273,13 @@ func (config *NetworkingTestConfig) DialFromNode(protocol, targetIP string, targ
...
-func (config *NetworkingTestConfig) GetSelfURL(port int32, path string, expected string) {
-	cmd := fmt.Sprintf("curl -i -q -s --connect-timeout 1 http://localhost:%d%s", port, path)
-	By(fmt.Sprintf("Getting kube-proxy self URL %s", path))
+// GetSelfStatusCode executes a curl against the given path via kubectl exec into a
+// test container running with host networking, and fails if the returned status
+// code doesn't match the expected string.
+func (config *NetworkingTestConfig) GetSelfURLStatusCode(port int32, path string, expected string) {
+	// check status code
+	cmd := fmt.Sprintf("curl -o /dev/null -i -q -s -w %%{http_code} --connect-timeout 1 http://localhost:%d%s", port, path)
+	By(fmt.Sprintf("Getting kube-proxy self URL %s status code", path))
```
Comment (bowei) [inline test/e2e/framework/networking_utils.go]: - This is not kube-proxy specific
- I think the other method still has usefulness

Comment (m1093782566) [inline test/e2e/framework/networking_utils.go]: oops, I will leave it and create a new function. THANKS for your reminding.

Comment (bowei) [inline test/e2e/framework/networking_utils.go]: The log message neds to be fixed...

**Comment by m1093782566** [comment]
@bowei review comments are fixed. PTAL.

**Review by cmluciano — COMMENTED** [review]
Is the existing [GetSelfURL](https://github.com/m1093782566/kubernetes/blob/9e54dfde70cb09d9fc837ea6b64b9fd1a0c251d0/test/e2e/networking.go#L85) insufficient for checking a 200 OK?

**Comment by m1093782566** [comment]
@cmluciano It's insufficient.

`GetSelfURL()` only return the response body. It works for `/healthz` endpoint since`/healthz` return `200 OK` as reponse body.

**Comment by cmluciano** [comment]
@m1093782566 Can you rebase this pr?

**Comment by cmluciano** [comment]
i'm not an official reviewer for this package. 

@lavalamp I think you are the one listed in the test_owners file. Can you take a look?

**Comment by bowei** [comment]
@m1093782566 you need to address the existing comments and to rebase your changes

File: test/e2e/framework/networking_utils.go (hunk 2 above)
Comment (m1093782566) [inline test/e2e/framework/networking_utils.go]: Oh sorry, I forgot it. Will fix it soon.

Comment (m1093782566) [inline test/e2e/framework/networking_utils.go]: By(fmt.Sprintf("Checking status code against %s", path))

Is that ok?

**Comment by m1093782566** [comment]
@bowei

Sorry for the delay, I just realized that I missed out one comment. I assume it's fixed now. Would 
 you please take a look again? Thanks!

**Comment by m1093782566** [comment]
ping @bowei?

**Comment by m1093782566** [comment]
Glad to receive your feedback, @bowei 

I create a help function `executeCmd(cmd string, expected string)`, and both `GetSeflURL()` and `GetSelfURLStatusCode()` will call it. 

Please take a look when you have a chance, thanks!

**Comment by m1093782566** [comment]
Thanks all ^_^

**Pull request merged**